tests/sentry/tasks/test_statistical_detectors.py

```diff
@@ -571,6 +631,66 @@ def test_detect_function_trends(
     assert detect_function_change_points.apply_async.called
 
 
+@mock.patch("sentry.tasks.statistical_detectors.functions.query")
+@mock.patch("sentry.tasks.statistical_detectors.detect_function_change_points")
+@mock.patch("sentry.statistical_detectors.detector.produce_occurrence_to_kafka")
+@django_db_all
+def test_detect_function_trends_auto_resolution(
+    produce_occurrence_to_kafka,
+    detect_function_change_points,
+    functions_query,
+    timestamp,
+    project,
+):
+    n = 750
+    timestamps = [timestamp - timedelta(hours=n - i) for i in range(n)]
+
+    functions_query.side_effect = [
+        {
+            "data": [
+                {
+                    "project.id": project.id,
+                    "fingerprint": 123,
+                    "count()": 100,
+                    "p95()": 100 if i < 10 else 300 if i < 20 else 100,
+                    "timestamp": ts.isoformat(),
+                },
+            ],
+        }
+        for i, ts in enumerate(timestamps)
+    ]
+
+    options = {
+        "statistical_detectors.enable": True,
+    }
+
+    features = {
+        "organizations:profiling-statistical-detectors-ema": [project.organization.slug],
+    }
+
+    with override_options(options), Feature(features):
+        for ts in timestamps[:20]:
+            detect_function_trends([project.id], ts)
+
+    assert detect_function_change_points.apply_async.called
+
+    with override_options(options), Feature(features):
+        RegressionGroup.objects.create(
+            type=RegressionType.FUNCTION.value,
+            date_regressed=timestamps[10],
+            version=1,
+            active=True,
+            project_id=project.id,
+            fingerprint=f"{123:x}",
+            baseline=100,
+            regressed=300,
+        )
+        for ts in timestamps[20:]:
+            detect_function_trends([project.id], ts)
+
+    assert produce_occurrence_to_kafka.called
```

Comment:
If we're going so far as to check that the occurrence is sent to kafka, should we also just add a check to know the correct status change is occurring?